src/modules/SuSEFirewall4Network.rb

```diff
@@ -218,28 +235,24 @@ def AnyInterfacesHandledByFirewall
     # @return	[Boolean] if successful
     def ProtectByFirewall(interface, zone, protect_status)
       # Adding protection
-      if protect_status == true
-        Builtins.y2milestone(
-          "Enabling firewall because of '%1' interface",
-          interface
-        )
-        SuSEFirewall.AddInterfaceIntoZone(interface, zone)
-        SuSEFirewall.SetEnableService(true)
-        SuSEFirewall.SetStartService(true)
-        # Removing protection
+      if protect_status
+        log.info("Enabling firewall because of '#{interface}' interface")
+
+        SuSEFirewall.AddInterfaceIntoZone(interface, zone) if !iface_in_zone?(interface, zone)
+
+        start_fw_service
```

Comment:
won't this set the dirty flag unnecessarily? maybe it should go inside the `if` block of `iface_in_zone?`?